lib/catalog/owid/catalog/datasets.py

```diff
@@ -205,14 +206,23 @@ def update_metadata(
             - "replace" (default): replace existing origin with new one
             - "append": append new origin to existing ones
             - "fail": raise an exception if origin already exists
+        :param errors: Optionally ignore errors like missing tables that are specified in YAML file
         """
         self.metadata.update_from_yaml(metadata_path, if_source_exists=if_source_exists)
 
         with open(metadata_path) as istream:
             metadata = yaml.safe_load(istream)
             for table_name in metadata.get("tables", {}).keys():
                 with disable_processing_log():
-                    table = self[table_name]
+                    try:
+                        table = self[table_name]
+                    except KeyError as e:
+                        if errors == "raise":
+                            raise e
+                        else:
+                            if errors == "warn":
+                                warnings.warn(str(e))
+                            continue
```

Comment:
What's the situation where this comes up? Is it because with a subset some tables are not included? Can you leave a comment behind explaining why you're making it more flexible here?

Reply:
> What's the situation where this comes up? Is it because with a subset some tables are not included?

Yes, this is useful if you work with `SUBSET` of dataset and have YAML metadata full of indicators. I'm updating the docs with motivation.